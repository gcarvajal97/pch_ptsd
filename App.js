import React from 'react';
import {
  StackNavigator,
} from 'react-navigation';
import {
  StyleSheet, 
  View,
  Button,
  Alert,
  Image,
} from 'react-native';

//TODO set up MVC so button press goes to new (empty) page
//TODO fix alignment of button and icon

export default function App() {
  return (
    <View style={styles.container}>
      <View style={styles.buttonBox}>
        <Image
          style={styles.image}
          source={require('./assets/healing_white_48dp_2x.png')}
          accessibilityLabel="Image of two band-aids criss-crossing"
        />
        <Button
          title="Learn About Trauma and Injury"
          onPress={() => Alert.alert('Go to Trauma and Injury Page')}
          style={styles.button}
          accessibilityLabel="Learn more about trauma and injury"
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    textAlign: 'center',
  },
  button: {
    color: '#c3fdff',
    width: '20%',
  },
  buttonBox: {
    height: '30%',
    width: '90%',
    backgroundColor: '#64b5f6',
    borderColor: '#1e88e5',
    borderRadius:10,
  },
  image: {
    flex: 1,
    alignItems: 'center',
    height: '40%',
    aspectRatio: 1,
  }
});
