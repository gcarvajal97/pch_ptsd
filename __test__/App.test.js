import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
jest.mock('../components/MainNavigator');
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe("App", () => {
    
    afterEach(() => {
        jest.clearAllMocks();
      });

    it("matches Snapshot", () => {
        const snap = renderer.create(<App />).toJSON();
        expect(snap).toMatchSnapshot();
    });
});