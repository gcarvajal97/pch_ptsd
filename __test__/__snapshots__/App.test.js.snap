// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App hides the splash screen and renders the app when the font is loaded 1`] = `ShallowWrapper {}`;

exports[`App matches Snapshot 1`] = `null`;
